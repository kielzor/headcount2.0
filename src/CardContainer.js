import React from 'react'
import Card from './Card'

const CardContainer = ({data, addCompareCards, evaluateCompareCard, comparedArr}) => {
  const updateClass = (cardData) => {
    if ((comparedArr[0] && cardData.location === comparedArr[0]['location']) ||
        (comparedArr[1] && cardData.location === comparedArr[1]['location'])) {
      return true
    }
    return false
  }

  const displayCards = data.map((cardData, i) =>
    <Card 
      evaluateCompareCard={evaluateCompareCard}
      data={cardData}
      addCompareCards={addCompareCards}
      key={`${cardData.location} + ${i}`}
      comparedInClass={updateClass(cardData)}
      compArrFull={comparedArr.length === 2 ? true : false}
    />
  )

  return (
    <div className='card-container'>
      {displayCards}
    </div>
  )
}

export default CardContainer