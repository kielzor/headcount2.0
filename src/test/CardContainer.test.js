import React from 'react'
import { shallow, mount } from 'enzyme'
import CardContainer from '../CardContainer'

describe('CardContainer', () => {
  let wrapper
  let mockData
  let mockCompare

  beforeEach(() => {
    mockData = []
    const mockUpdate = jest.fn()
    wrapper = shallow(<CardContainer 
                      data={mockData}
                      comparedArr={mockCompare}/>)
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should return information to the card', () => {
    expect(wrapper.find('div').length).toEqual(1)
  })

  // it('returns false if the card locations do not match', () => {
  //   mockData = [{location: 'COLORADO'}]
  //   mockCompare = [{target: {location: 'ACADEMY 20'}}]


  //   // wrapper = mount(<CardContainer 
  //   //                   data={mockData}
  //   //                   comparedArr={mockCompare}/>)


  //   console.log(wrapper.instance())
  //   const instance = wrapper.instance()

  //   instance.updateClass()


  //   expect(mockUpdate).toHaveBeenCalled()
  // })
})