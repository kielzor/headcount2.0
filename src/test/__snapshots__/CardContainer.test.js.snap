// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    comparedArr={undefined}
    data={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
